src/modalities/dataloader/large_file_lines_reader.py

```diff
@@ -41,6 +55,13 @@ def __init__(self, raw_data_path: Path, index_path: Optional[Path] = None):
         with self.index_path.open("rb") as f:
             self.index = pickle.load(f)
 
+        self.raw_data_fd = self.raw_data_path.open("rb")
+        self.mmapped_data_file = mmap.mmap(self.raw_data_fd.fileno(), 0, access=mmap.ACCESS_READ)
```

Comment:
The __init__-method opens these files, but there is no guarantee that close() is actually called. We could use a context manager (__enter__ and __exit__) for better resource management.

`def __enter__(self):
    return self

def __exit__(self, exc_type, exc_value, traceback):
    self.close()
`

Reply:
I created the issue https://github.com/Modalities/modalities/issues/271 for this.